Assets/HoloToolkit/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,343 @@
+            }
+
+            Interactibles.Add(interactible);
+        }
+
+#if UNITY_EDITOR
+        /// <summary>
+        /// When selected draw lines to all linked interactibles
+        /// </summary>
+        protected virtual void OnDrawGizmosSelected()
+        {
+            if (this.Interactibles.Count > 0)
+            {
+                GameObject[] bioList = this.Interactibles.ToArray();
+
+                for (int i = 0; i < bioList.Length; i++)
+                {
+                    if (bioList[i] != null)
+                    {
+                        Gizmos.color = Color.green;
+                        Gizmos.DrawLine(this.transform.position, bioList[i].transform.position);
+                    }
+                }
+            }
+
+            if (this.Targets.Count > 0)
+            {
+                GameObject[] targetList = this.Targets.ToArray();
+
+                for (int i = 0; i < targetList.Length; i++)
+                {
+                    if (targetList[i] != null)
+                    {
+                        Gizmos.color = Color.red;
+                        Gizmos.DrawLine(this.transform.position, targetList[i].transform.position);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// On Draw Gizmo show the receiver icon
+        /// </summary>
+        protected virtual void OnDrawGizmos()
+        {
+            _gizmoIcon = UnityEditor.Selection.activeGameObject == this.gameObject ? _gizmoIconSelected : _gizmoIconDefault;
+            Gizmos.DrawIcon(this.transform.position, _gizmoIcon, false);
+        }
+#endif
+
+        /// <summary>
+        /// Function to remove an interactible from the linked list.
+        /// </summary>
+        /// <param name="interactible"></param>
+        public virtual void RemoveInteractible(GameObject interactible)
+        {
+            if (interactible != null && Interactibles.Contains(interactible))
+            {
+                Interactibles.Remove(interactible);
+            }
+        }
+
+        /// <summary>
+        /// Clear the interactibles list and unregister them
+        /// </summary>
+        public virtual void ClearInteractibles()
+        {
+            GameObject[] _intList = Interactibles.ToArray();
+
+            for (int i = 0; i < _intList.Length; i++)
+            {
+                this.RemoveInteractible(_intList[i]);
+            }
+        }
+
+        /// <summary>
+        /// Is the game object interactible in our list of interactibles
+        /// </summary>
+        /// <param name="interactible"></param>
+        /// <returns></returns>
+        protected bool IsInteractible(GameObject interactible)
+        {
+            return (Interactibles != null && Interactibles.Contains(interactible));
+        }
+
+        private void CheckLockFocus(IPointingSource focuser)
+        {
+            if (bLockFocus)
+            {
+                //LockFocus(focuser);
```

Comment:
Why is this commented out?

Reply:
Need to add the functionality to the focus manager in MRTK.  On the HUX side individual focusers can have focus on an object locked which is different than modal as you can interact with anything else you like but focus is locked to an object for that focuser.